src/controllers/commands/CommandController.hpp

```diff
@@ -46,10 +47,16 @@ class CommandController final : public Singleton
     using CommandFunction =
         std::function<QString(QStringList /*words*/, ChannelPtr /*channel*/)>;
 
-    void registerCommand(QString commandName, CommandFunction commandFunction);
+    using CommandFunctionWithContext = std::function<QString(CommandContext)>;
+
+    using CommandFunctionAlternatives =
```

Comment:
Up to you but I feel like the name `CommandFunctionVariants` would make it easier to understand without looking at it's definition. Really means the same thing so not really a big deal either way.